File: week-03/day-3/ClassesAndFields/src/pirates/Pirate.java
package pirates;

import java.util.Random;

public class Pirate {
    int levelOfIntoxication;
    boolean isDead;

    public Pirate(){
        levelOfIntoxication = 0;
        isDead = false;
    }
    public void drinkSomeRum(){
        if (!isDead){
            levelOfIntoxication ++;
        } else {
            System.out.println("He is dead.");
        }
    }
    public void howItGoingMate(){
        if (0 <= levelOfIntoxication && levelOfIntoxication <= 4 && !isDead){
            System.out.println("Pour me anudder!");
        }else if(levelOfIntoxication > 4 && !isDead) {
            System.out.println("Arghh, I'ma Pirate. How d'ya d'ink its goin?");
        }else if(isDead){
            System.out.println("He is dead.");
        }
    }

    public void die(){
        isDead = true;
    }
    public boolean getIsDead(){
        return isDead;
    }

    public void brawl(Pirate pirateToBrawlWith){
        int chance = (int)(Math.random()*3);

        if(chance < 1 && !pirateToBrawlWith.isDead){
            this.die();
          }else if (chance >= 1 && chance < 2 && !pirateToBrawlWith.isDead){
            pirateToBrawlWith.die();
        }else if (chance >= 2 && !pirateToBrawlWith.isDead){
            this.die();
            pirateToBrawlWith.die();
        }else{
            System.out.println("He is dead!");
        }
    }
}

File: project_phase/day-4/Pirates/Pirates/Pirate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Pirates
{
    class Pirate
    {
        public string Name { get; set; }
        public int LevelOfIntoxication { get; set; }
        public bool IsSleeping { get; set; }
        public bool IsDead { get; set; }
        public bool IsCaptain { get; set; }

        public Pirate()
        {
            this.IsCaptain = false;
            this.IsSleeping = false;
            this.IsDead = false;
            this.LevelOfIntoxication = 0;
        }

        public T DoSomething<T>(dynamic input, dynamic input2)
        {
            return input + input2;
        }

        public Pirate(string name, bool isCaptain)
        {
            this.IsSleeping = false;
            this.IsDead = false;
            this.LevelOfIntoxication = 0;
            this.Name = name;
            this.IsCaptain = isCaptain;
        }

        public void DrinkSomeRum()
        {
            if (IsDead)
            {
                Console.WriteLine("he is dead");
            }
            else
            {
                this.LevelOfIntoxication++;
            }
        }

        public void HowItIsGoingMate()
        {
            if (IsDead)
            {
                Console.WriteLine("he is dead");
            }
            else
            {
                if (this.LevelOfIntoxication <= 4)
                {
                    Console.WriteLine("Pour me anudder!");
                }
                else
                {
                    Console.WriteLine($"Arghh, I'm Pirate. How d'ya d'ink its goin?");
                }
            }
        }

        public void Die()
        {
            this.IsDead = true;
        }

        public void Brawl(Pirate pirate)
        {
            if (this.IsDead)
            {
                Console.WriteLine("he is dead");
            }
            else
            {
                if (!pirate.IsDead)
                {
                    Random rnd = new Random();
                    int random = rnd.Next(1, 3);
                    if (random == 1)
                    {
                        this.IsDead = true;
                    }
                    if (random == 2)
                    {
                        pirate.IsDead = true;
                    }
                    if (random == 3)
                    {
                        this.IsSleeping = true;
                        pirate.IsSleeping = true;
                    }
                }
            }
        }
    }
}
